multiqc/modules/picard/ValidateSamFile.py

```diff
@@ -0,0 +1,369 @@
+    'MATES_ARE_SAME_END': 'Both mates of a pair are marked either as first or second mates',
+    'MISMATCH_FLAG_MATE_NEG_STRAND': 'Mate negative strand flag does not match read strand flag',
+    'MISMATCH_FLAG_MATE_UNMAPPED': 'Mate unmapped flag does not match read unmapped flag of mate',
+    'MISMATCH_MATE_ALIGNMENT_START': 'Mate alignment does not match alignment start of mate',
+    'MISMATCH_MATE_CIGAR_STRING': 'The mate cigar tag does not match its mate\'s cigar string',
+    'MISMATCH_MATE_REF_INDEX': 'Mate reference index (MRNM) does not match reference index of mate',
+    'MISMATCH_READ_LENGTH_AND_E2_LENGTH': 'Lengths of secondary base calls tag values and read should match',
+    'MISMATCH_READ_LENGTH_AND_QUALS_LENGTH': 'Length of sequence string and length of base quality string do not match',
+    'MISMATCH_READ_LENGTH_AND_U2_LENGTH': 'Secondary base quals tag values should match read length',
+    'MISSING_HEADER': 'The SAM/BAM file is missing the header',
+    'MISSING_PLATFORM_VALUE': 'The read group is missing its PL (platform unit) field',
+    'MISSING_READ_GROUP': 'The header is missing read group information',
+    'MISSING_SEQUENCE_DICTIONARY': 'There is no sequence dictionary in the header',
+    'MISSING_VERSION_NUMBER': 'Header has no version number',
+    'POORLY_FORMATTED_HEADER_TAG': 'Header tag does not have colon',
+    'READ_GROUP_NOT_FOUND': 'A read group ID on a SAMRecord is not found in the header',
+    'RECORD_OUT_OF_ORDER': 'The record is out of order',
+    'TAG_VALUE_TOO_LARGE': 'Unsigned integer tag value is deprecated in BAM. Template length',
+    'TRUNCATED_FILE': 'BAM file does not have terminator block',
+    'UNRECOGNIZED_HEADER_TYPE': 'Header record is not one of the standard types',
+}
+
+
+def _default_data_entry():
+    return {'WARNING_count': 0, 'ERROR_count': 0, 'file_validation_status': 'pass'}
+
+
+def parse_reports(self):
+    """
+        Find Picard ValidateSamFile reports and parse their data based on wether we
+        think it's a VERBOSE or SUMMARY report
+    """
+
+    # Get data
+    data = _parse_reports_by_type(self)
+
+    if data:
+        #  Filter to strip out ignored sample names (REQUIRED)
+        data = self.ignore_samples(data)
+
+        # Populate the general stats table
+        _add_data_to_general_stats(self, data)
+
+        # Add any found data to the report
+        _add_section_to_report(self, data)
+
+        # Write parsed data to a file
+        self.write_data_file(data, 'multiqc_picard_validatesamfile')
+
+    self.picard_ValidateSamFile_data = data  # Seems like the right thing to do
+    return len(data)
+
+
+def _parse_reports_by_type(self):
+    """ Returns a data dictionary
+
+        Goes through logs and parses them based on 'No errors found', VERBOSE or SUMMARY type.
+    """
+
+    data = dict()
+
+    for f in self.find_log_files('picard/sam_file_validation', filehandles=True):
+        sample = f['s_name']
```

Comment:
Do these files usually have `.SUMMARY` and `.VERBOSE` appended? These can be added to the `fn_clean_trim` config list if so to remove automatically.

Reply:
No, these files have no default filename what so ever.

The suffixes have been added:
*  In capitals because of a '.summary' search pattern of a different module.
* As an easy reference to keep assit debugging summary and verbose parsing.